interfaces/connection.go

```diff
@@ -45,6 +46,39 @@ type ConnectedSlot struct {
 	dynamicAttrs map[string]interface{}
 }
 
+// AttrGetter is an interface with Attr getter method common
+// to ConnectedSlot, ConnectedPlug, PlugInfo and SlotInfo types.
+type AttrGetter interface {
+	Attr(key string) (interface{}, error)
+}
+
+func getAttribute(staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, val interface{}) error {
+	var v interface{}
+	var ok bool
+
+	if dynamicAttrs != nil {
+		v, ok = dynamicAttrs[key]
+	}
+	if !ok && staticAttrs != nil {
+		v, ok = staticAttrs[key]
+	}
+
+	if ok {
+		rt := reflect.TypeOf(val)
+		if rt.Kind() != reflect.Ptr || val == nil {
+			return fmt.Errorf("cannot store the value of attribute %q", key)
```

Comment:
This error message does not reflect the action being done. Neither the developer nor the user at the other end have asked to store an attribute. Should be something along the lines of:

_"internal error: cannot get attribute with non-pointer value"_

Reply:
Similarly, we'd benefit from some context. Which interface type, which attribute. Otherwise we'll need to blindly hunt for the error when someone reports this.